assets/pages/new-contributions-landing/components/SetPassword/SetPassword.jsx

```diff
@@ -0,0 +1,154 @@
+  password: string,
+  guestAccountCreationToken: string,
+  setThankYouPageStage: (ThankYouPageStage) => void,
+  setPasswordHasBeenSubmitted: () => void,
+  passwordHasBeenSubmitted: boolean,
+  updatePassword: (Event) => void,
+  csrf: CsrfState,
+};
+/* eslint-enable react/no-unused-prop-types */
+
+
+// ----- State Maps ----- //
+
+const mapStateToProps = state => ({
+  contributionType: state.page.form.contributionType,
+  email: state.page.form.formData.email,
+  password: state.page.form.setPasswordData.password,
+  passwordHasBeenSubmitted: state.page.form.setPasswordData.passwordHasBeenSubmitted,
+  guestAccountCreationToken: state.page.form.guestAccountCreationToken,
+  csrf: state.page.csrf,
+});
+
+function mapDispatchToProps(dispatch: Dispatch<Action>) {
+  return {
+    setThankYouPageStage: (thankYouPageStage: ThankYouPageStage) => {
+      dispatch(setThankYouPageStage(thankYouPageStage));
+    },
+    setPasswordHasBeenSubmitted: () => {
+      dispatch(setPasswordHasBeenSubmitted());
+    },
+    updatePassword: (event: Event) => {
+      if (event.target instanceof HTMLInputElement) {
+        dispatch(updatePassword(event.target.value));
+      }
+    },
+  };
+}
+
+
+// ----- Functions ----- //
+
+function onSubmit(props: PropTypes): Event => void {
+  return (event) => {
+    props.setPasswordHasBeenSubmitted();
+    event.preventDefault();
+
+    if (!(event.target: any).checkValidity()) {
+      return;
+    }
+
+    // TODO: send user to thank you page with error if password was not set
+    setPasswordGuest(props.password, props.guestAccountCreationToken, props.csrf)
+      .then((response) => {
+        if (response === true) {
+          props.setThankYouPageStage('thankYouPasswordSet');
+        } else {
+          props.setThankYouPageStage('thankYouPasswordNotSet');
+        }
+      });
+  };
+}
+
+
+// ----- Render ----- //
+
+function SetPassword(props: PropTypes) {
+  if (!props.guestAccountCreationToken || !props.email) {
+    props.setThankYouPageStage('thankYou');
+  } else {
+    return (
+      <div className="set-password__container">
+        <h1 className="header">Set up a free account to manage your payments</h1>
+        <section className="set-password">
+          <p className="blurb">
+            Thank you for a valuable contribution. As a contributor, being signed in means you will no
+            longer see the “Since you’re here …” messages asking you to support our journalism.
+          </p>
+          <form onSubmit={onSubmit(props)} className={classNameWithModifiers('form', ['contribution'])} noValidate>
+            <NewContributionTextInput
+              id="email"
+              name="contribution-email"
+              label="Email address"
+              value={props.email}
+              isValid={checkEmail(props.email)}
+              pattern={emailRegexPattern}
+              icon={<SvgEnvelope />}
+              autoComplete="off"
+              autoCapitalize="words"
```

Comment:
for email addresses, I'd tend to enable auto complete and disable capitalisation.

Reply:
just saw
> This field doesn't actually ever get validated as it is fixed on this screen

you can then get rid of some attributes, can't you?

Reply:
`for email addresses, I'd tend to enable auto complete and disable capitalisation.` 👍 

`you can then get rid of some attributes, can't you?` - well, maybe it is safe to not assume it will be filled. We don't want to submit the form if the email isn't valid. If I don't set them, they will default to accepting anything, which maybe is worse. I'm not sure what I prefer tbh